src/qibo/tomography_RGD/qibo_states.py

```diff
@@ -0,0 +1,257 @@
+        Executes measurement circuits
+        - labels: string of Pauli matrices (e.g. XYZXX)
+        - backend: 'numpy', 'qibojit', 'pytorch', 'tensorflow' prvided by qibo
+        - num_shots: number of shots measurement is taken to get empirical frequency through counts
+        """
+        if self.measurement_circuit_names == []:
+            self.create_measurement_circuits(labels, label_format)
+
+        circuit_names = self.measurement_circuit_names
+
+        qibo.set_backend(backend)
+
+        data_dict_list = []
+        for i, label in enumerate(labels):
+            result = self.measurement_circuits[i].execute(nshots=num_shots)
+            result.samples(binary=True, registers=False)
+            count_dict = self.sort_sample(result.to_dict()["samples"])
+
+            measurement_circuit_name = self.make_measurement_circuit_name(
+                self.circuit_name, label
+            )
+            data_dict = {
+                "measurement_circuit_name": measurement_circuit_name,
+                "circuit_name": self.circuit_name,
+                "label": label,
+                "count_dict": count_dict,
+                "backend": "qibo: " + backend,
+                "num_shots": num_shots,
+            }
+            data_dict_list.append(data_dict)
+        return data_dict_list
+
+
+class GHZState(State):
+    """
+    Constructor for GHZState class
+    """
+
+    def __init__(self, n):
+        State.__init__(self, n)
+        self.circuit_name = "GHZ"
+
+    def create_circuit(self):
+        circuit = qibo.Circuit(self.n)
+
+        circuit.add(gates.H(0))
+        for i in range(1, self.n):
+            circuit.add(gates.CNOT(0, i))
+
+        self.circuit = circuit
+
+
+class HadamardState(State):
+    """
+    Constructor for HadamardState class
+    """
+
+    def __init__(self, n):
+        State.__init__(self, n)
+        self.circuit_name = "Hadamard"
+
+    def create_circuit(self):
+        circuit = qibo.Circuit(self.n)
+
+        for i in range(self.n):
+            circuit.add(gates.H(i))
+
+        self.circuit = circuit
+
+
+class RandomState(State):
+    """
+    Constructor for RandomState class
+    """
+
+    def __init__(self, n, seed=0, depth=40):
+        State.__init__(self, n)
+        self.circuit_name = "Random-%d" % (self.n,)
+
+        self.seed = seed
+        self.depth = depth
+
+    def create_circuit(self):
+        random.seed(a=self.seed)
+        circuit = qibo.Circuit(self.n)
+
+        for j in range(self.depth):
+            if self.n == 1:
+                op_ind = 0
+            else:
+                op_ind = random.randint(0, 1)
+            if op_ind == 0:  # U3
+                qind = random.randint(0, self.n - 1)
+                circuit.add(
+                    gates.U3(
+                        qind,
+                        2 * np.pi * random.random(),
+                        2 * np.pi * random.random(),
+                        2 * np.pi * random.random(),

```

Comment:
A tip for the future: This choice of phases does not sample random states uniformly. Please see https://github.com/qiboteam/qibo/blob/958b11bc38b800f217ad9c97b950635896935fd4/src/qibo/quantum_info/random_ensembles.py#L48 and https://pennylane.ai/qml/demos/tutorial_haar_measure